core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -79,73 +84,61 @@
    */
   public DefaultBlockWorkerClient(Subject subject, SocketAddress address) throws IOException {
     try {
-      mChannel = GrpcChannelBuilder.forAddress(address).setSubject(subject)
-          .setChannelType(NettyUtils.getClientChannelClass(!(address instanceof InetSocketAddress)))
-          .setEventLoopGroup(WORKER_GROUP)
-          .setKeepAliveTime(KEEPALIVE_TIME_MS, TimeUnit.MILLISECONDS)
-          .setKeepAliveTimeout(KEEPALIVE_TIMEOUT_MS, TimeUnit.MILLISECONDS)
-          .setMaxInboundMessageSize((int) MAX_INBOUND_MESSAGE_SIZE)
-          .setFlowControlWindow((int) GRPC_FLOWCONTROL_WINDOW).build();
+      mStreamingChannel = buildChannel(subject, address,
+          GrpcManagedChannelPool.PoolingStrategy.DISABLED);
+      mRpcChannel = buildChannel(subject, address,
+          GrpcManagedChannelPool.PoolingStrategy.DEFAULT);
```

Comment:
Is it the case we will always need to have these channels together? Otherwise we might be able to get away with not creating one of the channels.

Reply:
We don't always need both of them. I don't think we need to lazy init the channels as gRPC makes connection only at the first RPC call.

Reply:
Ah okay, that's good to know